src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
```

Comment:
You'r using same variable name twice. Better to change one instance to other name. E.g 'removeKey'.

Reply:
> You'r using same variable name twice. Better to change one instance to other name. E.g 'removeKey'.

Ah, ok, that's a good point, thank you.